packages/@glimmer/runtime/lib/compiled/opcodes/expressions.ts

```diff
@@ -123,14 +140,9 @@ APPEND_OPCODES.add(Op.HasBlockParams, vm => {
   // FIXME(mmun): should only need to push the symbol table
   let block = vm.stack.pop();
   let scope = vm.stack.pop();
-  check(block, CheckOption(CheckOr(CheckHandle, CheckCompilableBlock)));
-  check(scope, CheckOption(CheckScope));
-  let table = check(vm.stack.pop(), CheckOption(CheckBlockSymbolTable));
-
-  assert(
-    table === null || (table && typeof table === 'object' && Array.isArray(table.parameters)),
-    stackAssert('Option<BlockSymbolTable>', table)
-  );
+  check(block, CheckMaybe(CheckOr(CheckHandle, CheckCompilableBlock)));
+  check(scope, CheckMaybe(CheckScope));
+  let table = check(vm.stack.pop(), CheckMaybe(CheckBlockSymbolTable));
```

Comment:
Can we add a test case that does:

```hbs
<Foo as |bar|>{{bar}}</Foo>
```

Inside `{{bar}}` we say `{{has-block-params}}` (this was _probably_ broken prior to these changes, based on the comment just above in `HasBlock`).